docs/en/index.rst

```diff
@@ -85,6 +85,13 @@ Create a Job class::
     {
         use LogTrait;
 
+        /**
+         * The maximum number of times the job may be attempted.
+         * 
+         * @var int|null
+         */
+        public static $maxAttempts = 3;
```

Comment:
I think we should use a public static method instead of property. Methods allow users to add more custom logic should they need it and for simple scenarios returning a scalar value is not much code.

Reply:
Ok, are you ok with this being a required method and adding this to `JobInterface`? I didn't want to break compatibility with existing jobs that may not define this property so I am currently assuming null if it is not set. Are you envisioning the method taking arguments like the job data array? I don't quite see what custom logic would be possible without taking any arguments, but by passing something like the job data I can see some possibilities.

Reply:
> Ok, are you ok with this being a required method and adding this to JobInterface? I didn't want to break compatibility with existing jobs that may not define this property so I am currently assuming null if it is not set. 

Hmm, I wouldn't want to break compatibility for a feature that is opt-in. If we didn't have the method on the interface though we'd need to use `method_exists()` which isn't great either. I'm thinking the property might be better than a method now 🤔 

>  I don't quite see what custom logic would be possible without taking any arguments, but by passing something like the job data I can see some possibilities.

Developers would still have access to global state. Having the job data could give more options.